docs/client/session.md

```diff
@@ -0,0 +1,25 @@
+# Semantic conventions for session
+This document defines semantic conventions to apply to client-side applications when tracking sessions.
+
+A session is collection of Logs, Events, and Spans associated with a specific device or user within a specific period of time. 
+A session is represented by a unique identifier that is attached to Logs, Events, and Spans as an attribute.
+
+The primary purpose of sessions is to provide insight into the series of user actions or events that lead a critical or 
+interesting occurrence in the application's use. Sessions also provide a means to quantify overall application usage.
+
+
+## Attributes
+
+This attribute can be applied to all signals generated by an agent. It may be omitted from metrics, due to their aggregated nature. 
+
+| Attribute   | Type    | Description | Examples                             | Requirement level | 
+|-------------|---------|-------------|--------------------------------------|-------------------|
+| session.id  |  String | A GUID      | 00112233-4455-6677-8899-aabbccddeeff | required          |
+
+## How a session operates
+After a period of timeout, the `session.id` MUST be refreshed.
```

Comment:
I feel this is too prescriptive for me at the spec level. There are use cases where a session can just be the lifetime of the process (e.g. IoT), and having this requirement seems burdensome

Reply:
It seems like a session would be irrelevant to IoT if the use-case was to be the lifetime of the process. I would see sessions being useful for IoT if they captured discrete interactions with the devices. We should probably discuss IoT more throughly; I'm not extremely well versed in the potential use-cases.

Reply:
So even without IoT, there is the use console application use case (like a Point of Sales app like McDonald's or something), where the app is always on and there is long periods where there no activity, but you don't want to time that session out, or just send a ping to prevent a timeout.

Reply:
If no spans, logs, or events are recorded before the timeout period occurs then the next occurrence of one of those signals would trigger a session id refresh, thus starting a new session. Depending on the timeout period, this may end up with a session containing one checkout, or possibly a busy period with multiple checkouts. 

Streaming services may set their timeout to the average length of a movie, or generate a "ping" while a movie is being viewed, but that may not be necessary, as instrumented network requests for buffering would refresh the timeout through the viewing of content.

Reply:
Yeah, the new session triggering after some period of inactivity is the bit I'm not sure of. Perhaps we can support a value of 0 or something that indicates there's no timeout, so folks have to activity think about it in order to use that?

Reply:
I think that this is not necessary to be in the specification.

Using session identifier with OTel makes sense only for applications, that itself deploy some session (discrete client state) concept in its business logic, and want to use in analytics/telemetry. 

The specs purpose should be, IMHO, to enable associating this internal session with OTel signals. OTel spec should not assume much about the session lifecycle details itself. To be meaningful, the state is reset from time to time as dictated by the app business logic.

Let's analyse the McDonald's POS example. 

One can assume, the app there has some internal state that clears with each customer. From the analytics/telemetry point of view, it makes sense to associate the OTel session with this state. 

The state has its clear start (the POS itself clears the state at some moment as not to fuse the previous customers data with the new customer ones). 

The session on the POS ends from the business-loc point of view when the order is sent to backend systems. There, it can continue in the form of distributed traces, that originate on the POS terminal in the scope of the client session there (the session identifier can be distributed in baggage too, eventually). I.e., there can be session-associated signals traced long after the session itself (in the POS terminal) no longer exists. E.g., till the order is handled to the waiting customer.

What happens on the POS terminal in the idle time? 

First of all, are there some signals during the idle time, anyway? If not, it is not important whether the (previous) session identifier is cached or set somewhere or not, as it is no used. If there are some signals, then the logic step would probably be to clear the session identifier when the business-logic session ends in the POS terminal, and trace the idle-time signals as "session-less".